metrics-operator/api/v1alpha3/analysis_types_test.go

```diff
@@ -0,0 +1,193 @@
+package v1alpha3
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAnalysis_GetFromTime(t *testing.T) {
+	type fields struct {
+		TypeMeta   v1.TypeMeta
+		ObjectMeta v1.ObjectMeta
+		Spec       AnalysisSpec
+		Status     AnalysisStatus
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   time.Time
+	}{
+		// TODO: Add test cases.
```

Comment:
Can we add some test cases ? :)

Reply:
ah sure, that todo slipped through, good catch ;)